### ISSUE TITLE
llvm-c-test undefined references

### ISSUE THREAD

**Issue by kerberizer** [kerberizer/llvm-svn#1]
```
[ 84%] Linking C executable ../../bin/llvm-c-test
CMakeFiles/llvm-c-test.dir/main.c.o: In function `main':
main.c:(.text.startup.main+0xc): undefined reference to `LLVMGetGlobalPassRegistry'
main.c:(.text.startup.main+0x14): undefined reference to `LLVMInitializeCore'
CMakeFiles/llvm-c-test.dir/calc.c.o: In function `build_from_tokens':
calc.c:(.text.build_from_tokens+0x9e): undefined reference to `LLVMInt64Type'
calc.c:(.text.build_from_tokens+0xb2): undefined reference to `LLVMConstInt'
calc.c:(.text.build_from_tokens+0x119): undefined reference to `LLVMBuildGEP'
calc.c:(.text.build_from_tokens+0x12b): undefined reference to `LLVMBuildLoad'
calc.c:(.text.build_from_tokens+0x149): undefined reference to `LLVMBuildRet'
calc.c:(.text.build_from_tokens+0x1f3): undefined reference to `LLVMBuildBinOp'
CMakeFiles/llvm-c-test.dir/calc.c.o: In function `handle_line':
calc.c:(.text.handle_line+0x29): undefined reference to `LLVMModuleCreateWithName'
calc.c:(.text.handle_line+0x31): undefined reference to `LLVMInt64Type'
calc.c:(.text.handle_line+0x3e): undefined reference to `LLVMPointerType'
calc.c:(.text.handle_line+0x57): undefined reference to `LLVMFunctionType'
calc.c:(.text.handle_line+0x65): undefined reference to `LLVMAddFunction'
calc.c:(.text.handle_line+0x6d): undefined reference to `LLVMCreateBuilder'
calc.c:(.text.handle_line+0x7f): undefined reference to `LLVMAppendBasicBlock'
calc.c:(.text.handle_line+0x8a): undefined reference to `LLVMPositionBuilderAtEnd'
calc.c:(.text.handle_line+0x97): undefined reference to `LLVMGetParams'
calc.c:(.text.handle_line+0xa8): undefined reference to `LLVMSetValueName'
calc.c:(.text.handle_line+0xcb): undefined reference to `LLVMPrintModuleToString'
calc.c:(.text.handle_line+0xde): undefined reference to `LLVMDisposeMessage'
calc.c:(.text.handle_line+0xe6): undefined reference to `LLVMDisposeBuilder'
calc.c:(.text.handle_line+0xee): undefined reference to `LLVMDisposeModule'
CMakeFiles/llvm-c-test.dir/disassemble.c.o: In function `handle_line':
disassemble.c:(.text.handle_line+0xe7): undefined reference to `LLVMCreateDisasmCPUFeatures'
disassemble.c:(.text.handle_line+0x13e): undefined reference to `LLVMDisasmInstruction'
disassemble.c:(.text.handle_line+0x1c8): undefined reference to `LLVMDisasmDispose'
disassemble.c:(.text.handle_line+0x2a8): undefined reference to `LLVMCreateDisasmCPUFeatures'
CMakeFiles/llvm-c-test.dir/disassemble.c.o: In function `disassemble':
disassemble.c:(.text.disassemble+0x5): undefined reference to `LLVMInitializeAArch64TargetInfo'
disassemble.c:(.text.disassemble+0xa): undefined reference to `LLVMInitializeAMDGPUTargetInfo'
disassemble.c:(.text.disassemble+0xf): undefined reference to `LLVMInitializeARMTargetInfo'
disassemble.c:(.text.disassemble+0x14): undefined reference to `LLVMInitializeBPFTargetInfo'
disassemble.c:(.text.disassemble+0x19): undefined reference to `LLVMInitializeCppBackendTargetInfo'
disassemble.c:(.text.disassemble+0x1e): undefined reference to `LLVMInitializeHexagonTargetInfo'
disassemble.c:(.text.disassemble+0x23): undefined reference to `LLVMInitializeMipsTargetInfo'
disassemble.c:(.text.disassemble+0x28): undefined reference to `LLVMInitializeMSP430TargetInfo'
disassemble.c:(.text.disassemble+0x2d): undefined reference to `LLVMInitializeNVPTXTargetInfo'
disassemble.c:(.text.disassemble+0x32): undefined reference to `LLVMInitializePowerPCTargetInfo'
disassemble.c:(.text.disassemble+0x37): undefined reference to `LLVMInitializeSparcTargetInfo'
disassemble.c:(.text.disassemble+0x3c): undefined reference to `LLVMInitializeSystemZTargetInfo'
disassemble.c:(.text.disassemble+0x41): undefined reference to `LLVMInitializeX86TargetInfo'
disassemble.c:(.text.disassemble+0x46): undefined reference to `LLVMInitializeXCoreTargetInfo'
disassemble.c:(.text.disassemble+0x4b): undefined reference to `LLVMInitializeAArch64TargetMC'
disassemble.c:(.text.disassemble+0x50): undefined reference to `LLVMInitializeAMDGPUTargetMC'
disassemble.c:(.text.disassemble+0x55): undefined reference to `LLVMInitializeARMTargetMC'
disassemble.c:(.text.disassemble+0x5a): undefined reference to `LLVMInitializeBPFTargetMC'
disassemble.c:(.text.disassemble+0x5f): undefined reference to `LLVMInitializeCppBackendTargetMC'
disassemble.c:(.text.disassemble+0x64): undefined reference to `LLVMInitializeHexagonTargetMC'
disassemble.c:(.text.disassemble+0x69): undefined reference to `LLVMInitializeMipsTargetMC'
disassemble.c:(.text.disassemble+0x6e): undefined reference to `LLVMInitializeMSP430TargetMC'
disassemble.c:(.text.disassemble+0x73): undefined reference to `LLVMInitializeNVPTXTargetMC'
disassemble.c:(.text.disassemble+0x78): undefined reference to `LLVMInitializePowerPCTargetMC'
disassemble.c:(.text.disassemble+0x7d): undefined reference to `LLVMInitializeSparcTargetMC'
disassemble.c:(.text.disassemble+0x82): undefined reference to `LLVMInitializeSystemZTargetMC'
disassemble.c:(.text.disassemble+0x87): undefined reference to `LLVMInitializeX86TargetMC'
disassemble.c:(.text.disassemble+0x8c): undefined reference to `LLVMInitializeXCoreTargetMC'
disassemble.c:(.text.disassemble+0x91): undefined reference to `LLVMInitializeAArch64Disassembler'
disassemble.c:(.text.disassemble+0x96): undefined reference to `LLVMInitializeARMDisassembler'
disassemble.c:(.text.disassemble+0x9b): undefined reference to `LLVMInitializeHexagonDisassembler'
disassemble.c:(.text.disassemble+0xa0): undefined reference to `LLVMInitializeMipsDisassembler'
disassemble.c:(.text.disassemble+0xa5): undefined reference to `LLVMInitializePowerPCDisassembler'
disassemble.c:(.text.disassemble+0xaa): undefined reference to `LLVMInitializeSparcDisassembler'
disassemble.c:(.text.disassemble+0xaf): undefined reference to `LLVMInitializeSystemZDisassembler'
disassemble.c:(.text.disassemble+0xb4): undefined reference to `LLVMInitializeX86Disassembler'
disassemble.c:(.text.disassemble+0xb9): undefined reference to `LLVMInitializeXCoreDisassembler'
CMakeFiles/llvm-c-test.dir/module.c.o: In function `module_dump':
module.c:(.text.module_dump+0x2b): undefined reference to `LLVMCreateMemoryBufferWithSTDIN'
module.c:(.text.module_dump+0x40): undefined reference to `LLVMParseBitcode'
module.c:(.text.module_dump+0x4d): undefined reference to `LLVMDisposeMemoryBuffer'
module.c:(.text.module_dump+0x5a): undefined reference to `LLVMPrintModuleToString'
module.c:(.text.module_dump+0x6d): undefined reference to `LLVMDisposeMessage'
module.c:(.text.module_dump+0x75): undefined reference to `LLVMDisposeModule'
module.c:(.text.module_dump+0xe0): undefined reference to `LLVMDisposeMemoryBuffer'
CMakeFiles/llvm-c-test.dir/module.c.o: In function `module_list_functions':
module.c:(.text.module_list_functions+0x33): undefined reference to `LLVMCreateMemoryBufferWithSTDIN'
module.c:(.text.module_list_functions+0x4c): undefined reference to `LLVMParseBitcode'
module.c:(.text.module_list_functions+0x5d): undefined reference to `LLVMDisposeMemoryBuffer'
module.c:(.text.module_list_functions+0x6a): undefined reference to `LLVMGetFirstFunction'
module.c:(.text.module_list_functions+0x81): undefined reference to `LLVMGetValueName'
module.c:(.text.module_list_functions+0x9a): undefined reference to `LLVMGetNextFunction'
module.c:(.text.module_list_functions+0xae): undefined reference to `LLVMIsDeclaration'
module.c:(.text.module_list_functions+0xba): undefined reference to `LLVMCountBasicBlocks'
module.c:(.text.module_list_functions+0xc7): undefined reference to `LLVMGetValueName'
module.c:(.text.module_list_functions+0xe4): undefined reference to `LLVMGetFirstBasicBlock'
module.c:(.text.module_list_functions+0xf8): undefined reference to `LLVMGetFirstInstruction'
module.c:(.text.module_list_functions+0x10f): undefined reference to `LLVMIsACallInst'
module.c:(.text.module_list_functions+0x11c): undefined reference to `LLVMGetNumOperands'
module.c:(.text.module_list_functions+0x127): undefined reference to `LLVMGetOperand'
module.c:(.text.module_list_functions+0x12f): undefined reference to `LLVMGetValueName'
module.c:(.text.module_list_functions+0x148): undefined reference to `LLVMGetNextInstruction'
module.c:(.text.module_list_functions+0x158): undefined reference to `LLVMGetNextBasicBlock'
module.c:(.text.module_list_functions+0x189): undefined reference to `LLVMGetNextFunction'
module.c:(.text.module_list_functions+0x19d): undefined reference to `LLVMDisposeModule'
module.c:(.text.module_list_functions+0x20b): undefined reference to `LLVMDisposeMemoryBuffer'
CMakeFiles/llvm-c-test.dir/module.c.o: In function `module_list_globals':
module.c:(.text.module_list_globals+0x33): undefined reference to `LLVMCreateMemoryBufferWithSTDIN'
module.c:(.text.module_list_globals+0x4c): undefined reference to `LLVMParseBitcode'
module.c:(.text.module_list_globals+0x6b): undefined reference to `LLVMDisposeMemoryBuffer'
module.c:(.text.module_list_globals+0x78): undefined reference to `LLVMGetFirstGlobal'
module.c:(.text.module_list_globals+0x8c): undefined reference to `LLVMTypeOf'
module.c:(.text.module_list_globals+0x94): undefined reference to `LLVMPrintTypeToString'
module.c:(.text.module_list_globals+0x9f): undefined reference to `LLVMGetValueName'
module.c:(.text.module_list_globals+0xaa): undefined reference to `LLVMIsDeclaration'
module.c:(.text.module_list_globals+0xcf): undefined reference to `LLVMDisposeMessage'
module.c:(.text.module_list_globals+0xd7): undefined reference to `LLVMGetNextGlobal'
module.c:(.text.module_list_globals+0xe7): undefined reference to `LLVMDisposeModule'
module.c:(.text.module_list_globals+0x15a): undefined reference to `LLVMDisposeMemoryBuffer'
CMakeFiles/llvm-c-test.dir/metadata.c.o: In function `add_named_metadata_operand':
metadata.c:(.text.add_named_metadata_operand+0x1d): undefined reference to `LLVMModuleCreateWithName'
metadata.c:(.text.add_named_metadata_operand+0x25): undefined reference to `LLVMInt32Type'
metadata.c:(.text.add_named_metadata_operand+0x31): undefined reference to `LLVMConstInt'
metadata.c:(.text.add_named_metadata_operand+0x42): undefined reference to `LLVMMDNode'
metadata.c:(.text.add_named_metadata_operand+0x54): undefined reference to `LLVMAddNamedMetadataOperand'
metadata.c:(.text.add_named_metadata_operand+0x5c): undefined reference to `LLVMDisposeModule'
CMakeFiles/llvm-c-test.dir/metadata.c.o: In function `set_metadata':
metadata.c:(.text.set_metadata+0x19): undefined reference to `LLVMCreateBuilder'
metadata.c:(.text.set_metadata+0x21): undefined reference to `LLVMInt32Type'
metadata.c:(.text.set_metadata+0x2d): undefined reference to `LLVMConstInt'
metadata.c:(.text.set_metadata+0x3e): undefined reference to `LLVMMDNode'
metadata.c:(.text.set_metadata+0x52): undefined reference to `LLVMGetMDKindID'
metadata.c:(.text.set_metadata+0x5c): undefined reference to `LLVMBuildRetVoid'
metadata.c:(.text.set_metadata+0x69): undefined reference to `LLVMSetMetadata'
metadata.c:(.text.set_metadata+0x71): undefined reference to `LLVMDisposeBuilder'
CMakeFiles/llvm-c-test.dir/object.c.o: In function `object_list_sections':
object.c:(.text.object_list_sections+0x2e): undefined reference to `LLVMCreateMemoryBufferWithSTDIN'
object.c:(.text.object_list_sections+0x40): undefined reference to `LLVMCreateObjectFile'
object.c:(.text.object_list_sections+0x54): undefined reference to `LLVMGetSections'
object.c:(.text.object_list_sections+0x61): undefined reference to `LLVMGetSectionSize'
object.c:(.text.object_list_sections+0x6c): undefined reference to `LLVMGetSectionAddress'
object.c:(.text.object_list_sections+0x77): undefined reference to `LLVMGetSectionName'
object.c:(.text.object_list_sections+0x96): undefined reference to `LLVMMoveToNextSection'
object.c:(.text.object_list_sections+0xa1): undefined reference to `LLVMIsSectionIteratorAtEnd'
object.c:(.text.object_list_sections+0xad): undefined reference to `LLVMDisposeSectionIterator'
object.c:(.text.object_list_sections+0xb5): undefined reference to `LLVMDisposeObjectFile'
CMakeFiles/llvm-c-test.dir/object.c.o: In function `object_list_symbols':
object.c:(.text.object_list_symbols+0x32): undefined reference to `LLVMCreateMemoryBufferWithSTDIN'
object.c:(.text.object_list_symbols+0x44): undefined reference to `LLVMCreateObjectFile'
object.c:(.text.object_list_symbols+0x58): undefined reference to `LLVMGetSections'
object.c:(.text.object_list_symbols+0x63): undefined reference to `LLVMGetSymbols'
object.c:(.text.object_list_symbols+0x77): undefined reference to `LLVMMoveToContainingSection'
object.c:(.text.object_list_symbols+0x7f): undefined reference to `LLVMGetSectionName'
object.c:(.text.object_list_symbols+0x8a): undefined reference to `LLVMGetSymbolSize'
object.c:(.text.object_list_symbols+0x95): undefined reference to `LLVMGetSymbolAddress'
object.c:(.text.object_list_symbols+0xa0): undefined reference to `LLVMGetSymbolName'
object.c:(.text.object_list_symbols+0xc2): undefined reference to `LLVMMoveToNextSymbol'
object.c:(.text.object_list_symbols+0xcd): undefined reference to `LLVMIsSymbolIteratorAtEnd'
object.c:(.text.object_list_symbols+0xd9): undefined reference to `LLVMDisposeSymbolIterator'
object.c:(.text.object_list_symbols+0xe1): undefined reference to `LLVMDisposeObjectFile'
CMakeFiles/llvm-c-test.dir/targets.c.o: In function `targets_list':
targets.c:(.text.targets_list+0x2): undefined reference to `LLVMInitializeAArch64TargetInfo'
targets.c:(.text.targets_list+0x7): undefined reference to `LLVMInitializeAMDGPUTargetInfo'
targets.c:(.text.targets_list+0xc): undefined reference to `LLVMInitializeARMTargetInfo'
targets.c:(.text.targets_list+0x11): undefined reference to `LLVMInitializeBPFTargetInfo'
targets.c:(.text.targets_list+0x16): undefined reference to `LLVMInitializeCppBackendTargetInfo'
targets.c:(.text.targets_list+0x1b): undefined reference to `LLVMInitializeHexagonTargetInfo'
targets.c:(.text.targets_list+0x20): undefined reference to `LLVMInitializeMipsTargetInfo'
targets.c:(.text.targets_list+0x25): undefined reference to `LLVMInitializeMSP430TargetInfo'
targets.c:(.text.targets_list+0x2a): undefined reference to `LLVMInitializeNVPTXTargetInfo'
targets.c:(.text.targets_list+0x2f): undefined reference to `LLVMInitializePowerPCTargetInfo'
targets.c:(.text.targets_list+0x34): undefined reference to `LLVMInitializeSparcTargetInfo'
targets.c:(.text.targets_list+0x39): undefined reference to `LLVMInitializeSystemZTargetInfo'
targets.c:(.text.targets_list+0x3e): undefined reference to `LLVMInitializeX86TargetInfo'
targets.c:(.text.targets_list+0x43): undefined reference to `LLVMInitializeXCoreTargetInfo'
targets.c:(.text.targets_list+0x48): undefined reference to `LLVMInitializeAArch64Target'
targets.c:(.text.targets_list+0x4d): undefined reference to `LLVMInitializeAMDGPUTarget'
targets.c:(.text.targets_list+0x52): undefined reference to `LLVMInitializeARMTarget'
targets.c:(.text.targets_list+0x57): undefined reference to `LLVMInitializeBPFTarget'
targets.c:(.text.targets_list+0x5c): undefined reference to `LLVMInitializeCppBackendTarget'
targets.c:(.text.targets_list+0x61): undefined reference to `LLVMInitializeHexagonTarget'
targets.c:(.text.targets_list+0x66): undefined reference to `LLVMInitializeMipsTarget'
targets.c:(.text.targets_list+0x6b): undefined reference to `LLVMInitializeMSP430Target'
targets.c:(.text.targets_list+0x70): undefined reference to `LLVMInitializeNVPTXTarget'
targets.c:(.text.targets_list+0x75): undefined reference to `LLVMInitializePowerPCTarget'
targets.c:(.text.targets_list+0x7a): undefined reference to `LLVMInitializeSparcTarget'
targets.c:(.text.targets_list+0x7f): undefined reference to `LLVMInitializeSystemZTarget'
targets.c:(.text.targets_list+0x84): undefined reference to `LLVMInitializeX86Target'
targets.c:(.text.targets_list+0x89): undefined reference to `LLVMInitializeXCoreTarget'
targets.c:(.text.targets_list+0x8e): undefined reference to `LLVMGetFirstTarget'
targets.c:(.text.targets_list+0xa4): undefined reference to `LLVMGetTargetDescription'
targets.c:(.text.targets_list+0xbd): undefined reference to `LLVMGetNextTarget'
targets.c:(.text.targets_list+0xcd): undefined reference to `LLVMGetTargetName'
targets.c:(.text.targets_list+0xe6): undefined reference to `LLVMTargetHasJIT'
collect2: error: ld returned 1 exit status
tools/llvm-c-test/CMakeFiles/llvm-c-test.dir/build.make:303: recipe for target 'bin/llvm-c-test' failed
make[2]: *** [bin/llvm-c-test] Error 1
CMakeFiles/Makefile2:31281: recipe for target 'tools/llvm-c-test/CMakeFiles/llvm-c-test.dir/all' failed
make[1]: *** [tools/llvm-c-test/CMakeFiles/llvm-c-test.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

**Comment by kerberizer** [comment]
The reason most likely is abnormal libLLVM.so, as described in [this bug report](https://llvm.org/bugs/show_bug.cgi?id=23649).

**Comment by kerberizer** [comment]
Fixed by commit 4be3c74f7a687594e68e0a7b38aa159dbeb04b43.
